shared/src/domain-logic/opportunityFields.test.ts

```diff
@@ -0,0 +1,240 @@
+    );
+
+    it(
+      "returns only date of formation and municipality keys when it is a " +
+        "foreign business with a business type of nexus and a legal structure that is not a trade name and the location is in NJ",
+      async () => {
+        const foreignNexusBusinessWithoutLocationInNJ = generateProfileData({
+          businessPersona: "FOREIGN",
+          foreignBusinessType: "NEXUS",
+          nexusLocationInNewJersey: true,
+          legalStructureId: "limited-partnership",
+        });
+
+        const profileFields = getFieldsForProfile(foreignNexusBusinessWithoutLocationInNJ);
+        expect(profileFields).toContain("dateOfFormation");
+        expect(profileFields).toContain("municipality");
+        expect(profileFields).not.toContain("existingEmployees");
+        expect(profileFields).not.toContain("homeBasedBusiness");
+        expect(profileFields).not.toContain("ownershipTypeIds");
+      }
+    );
+
+    it(
+      "returns only municipality  when it is a " +
+        "foreign business with a business type of nexus and a legal structure that is a trade name and the location is in NJ",
+      async () => {
+        const foreignNexusBusinessWithoutLocationInNJ = generateProfileData({
+          businessPersona: "FOREIGN",
+          foreignBusinessType: "NEXUS",
+          nexusLocationInNewJersey: true,
+          legalStructureId: "general-partnership",
+        });
+
+        const profileFields = getFieldsForProfile(foreignNexusBusinessWithoutLocationInNJ);
+        expect(profileFields).toContain("municipality");
+        expect(profileFields).not.toContain("dateOfFormation");
+        expect(profileFields).not.toContain("existingEmployees");
+        expect(profileFields).not.toContain("homeBasedBusiness");
+        expect(profileFields).not.toContain("ownershipTypeIds");
+      }
+    );
+
+    it(
+      "returns only home based business when it is a " +
+        "foreign business with a business type of nexus and a legal structure that is a trade name and the location is not in NJ",
+      async () => {
+        const foreignNexusBusinessWithoutLocationInNJ = generateProfileData({
+          businessPersona: "FOREIGN",
+          foreignBusinessType: "NEXUS",
+          nexusLocationInNewJersey: false,
+          legalStructureId: "general-partnership",
+        });
+
+        const profileFields = getFieldsForProfile(foreignNexusBusinessWithoutLocationInNJ);
+        expect(profileFields).toContain("homeBasedBusiness");
+        expect(profileFields).not.toContain("municipality");
+        expect(profileFields).not.toContain("dateOfFormation");
+        expect(profileFields).not.toContain("existingEmployees");
+        expect(profileFields).not.toContain("ownershipTypeIds");
+      }
+    );
+
+    it(
+      "returns municipality, existing employees, home based business, and ownership type id keys when it is " +
+        "a starting business and the legal structure id is general-partnership",
```

Comment:
[pebble] the other tests say Trade Name, which is accurate. I think saying "when legal structure is general-partnership" is too specific for the behavior we want to describe

Reply:
changed to trade name